Find a functionally equivalent code:android.media.AudioService.sendVolumeUpdate:COMMENT
Method Modifier: private     
Comment:// UI update and Broadcast Intent

@@ -1,8 +1,16 @@
 {
-    if (!mVoiceCapable && (streamType == AudioSystem.STREAM_RING)) {
+    if (!isPlatformVoice() && (streamType == AudioSystem.STREAM_RING)) {
         streamType = AudioSystem.STREAM_NOTIFICATION;
     }
-    mVolumePanel.postVolumeChanged(streamType, flags);
+    // only when TV receives volume notification from Audio Receiver.
+    if (mHdmiTvClient != null && streamType == AudioSystem.STREAM_MUSIC) {
+        synchronized (mHdmiTvClient) {
+            if (mHdmiSystemAudioSupported && ((flags & AudioManager.FLAG_HDMI_SYSTEM_AUDIO_VOLUME) == 0)) {
+                flags &= ~AudioManager.FLAG_SHOW_UI;
+            }
+        }
+    }
+    mVolumeController.postVolumeChanged(streamType, flags);
     if ((flags & AudioManager.FLAG_FIXED_VOLUME) == 0) {
         oldIndex = (oldIndex + 5) / 10;
         index = (index + 5) / 10;

